vectordatasource/transform.py

```diff
@@ -4515,6 +4559,20 @@ def merge_networks_from_tags(shape, props, fid, zoom):
         # apply country-specific logic to try and backfill the network from
         # structure we know about how refs work in the country.
         logic = _COUNTRY_SPECIFIC_ROAD_NETWORK_LOGIC.get(country_code)
+
+        # if the road is a member of exactly one road relation, which provides
+        # a network and no ref, and the element itself provides no network,
+        # then use the network from the relation instead.
+        if network is None:
+            solo_networks_from_relations = []
+            for i in xrange(0, len(mz_networks), 3):
+                t, n, r = mz_networks[i:i+3]
+                if t == 'road' and n and r is None:
+                    solo_networks_from_relations.append((n, i))
+            if len(solo_networks_from_relations) == 1:
+                network, i = solo_networks_from_relations[0]
+                del mz_networks[i:i+3]
```

Comment:
Is this right? Seems that we would need to remember the index `i`, otherwise we could have skipped past it and just removed the last one? I'm assuming that the intent is to remove the solo network from `mz_networks`, but I could just be misunderstanding.

Reply:
The code only does the `del` if there's only one match. We won't be doing a second deletion, so `i` doesn't need to be adjusted to account for prior deletions.

Does that sound right?

Reply:
Oh I'm sorry, you are remembering the index in the `solo_networks_from_relations` variable. I somehow missed where you saved it, *and* where you pulled it out. Carry on! :)

Reply:
Added a better comment in 3a56e2a.